probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -1059,16 +1059,18 @@ impl CoreInterface for Armv7a<'_> {
         if let Some(endianness) = self.endianness {
             return Ok(endianness);
         }
-        let endianness = {
-            let psr = TryInto::<u32>::try_into(self.read_core_reg(XPSR.id())?).unwrap();
-            if psr & 1 << 9 == 0 {
-                Endian::Little
-            } else {
-                Endian::Big
-            }
-        };
-        self.endianness = Some(endianness);
-        Ok(endianness)
+        self.halted_access(|core| {
+            let endianness = {
+                let psr = TryInto::<u32>::try_into(core.read_core_reg(XPSR.id())?).unwrap();
```

Comment:
Just one more thing: on xtensa, we need to halt the core for basically all operations, but certainly for each `read_core_reg`. As `halted_access` is basically a no-op if already halted, maybe you'd want to wrap the implementation of `read_core_reg`, so you don't have to find each call site and halt around them.

Reply:
All architectures likely need to have the core halted to call `read_core_reg`, so I don't see the need to wrap the entire function like that, unless we want to do that for all platforms.